src/bootstrap/check.rs

```diff
@@ -321,11 +321,9 @@ macro_rules! tool_check_step {
                 }
 
                 // Enable internal lints for clippy and rustdoc
```

Comment:
Uh, this comment seems odd or at least in need of updates - mind removing it I guess?

Are we gating internal lints on a *CLI* flag being passed? I guess we're enabling them on the CLI so that makes some sense, but then I'm not sure why this piece of code is in check.rs vs. Builder::cargo or so, alongside actually enabling the lints.

Generally deny(rustc::internal) isn't quite what we want in-code too, as we want deny-warnings=false to work locally...

Reply:
> Are we gating internal lints on a CLI flag being passed?

Yes, the compiler doesn't run internal lints unless -Z unstable-options is passed, even if you have `#[warn(rustc::internal)]`.

> Generally deny(rustc::internal) isn't quite what we want in-code too, as we want deny-warnings=false to work locally...

Oh good point - I'll fix it in rustdoc and update the comment.

Reply:
> Oh good point - I'll fix it in rustdoc and update the comment.

Done.